main/src/ca/uwaterloo/flix/language/phase/typer/ConstraintGen.scala

```diff
@@ -439,25 +439,18 @@ object ConstraintGen {
 
       case e: Expr.RestrictableChoose => RestrictableChooseConstraintGen.visitRestrictableChoose(e)
 
-      case Expr.ExtMatch(label, exp1, sym2, exp2, sym3, exp3, tvar, loc) =>
-        val pred = Name.Pred(label.name, label.loc)
-
-        val (tpe1, eff1) = visitExp(exp1)
-        val (tpe2, eff2) = visitExp(exp2)
-        val (tpe3, eff3) = visitExp(exp3)
-
-        val freshTypeVar = freshVar(Kind.Star, loc)
+      case Expr.ExtMatch(exp, rules, loc) =>
+        val (tpe, eff) = visitExp(exp)
+        val (caseTpes, tpes, effs) = rules.map(visitExtMatchRule).unzip3
         val freshRowVar = freshVar(Kind.SchemaRow, loc)
-        val expectedRowType = Type.mkSchemaRowExtend(pred, Type.mkRelation(List(freshTypeVar), loc), freshRowVar, loc)
+        val expectedRowType = caseTpes.foldRight(freshRowVar: Type) {
```

Comment:
`freshRowVar` here should instead be `EmptySchemaRow` since we are only considering closed rows currently.

Reply:
Yes, if the scrutinee is too big, then we'll get a type error, but that is OK because it indicates an incomplete match.

If the scrutinee is too small, then we'll also get a type error, but that is (maybe) OK because it indicates an unreachable case.

Reply:
As Matthew said